### PR TITLE
VizTooltip: Show data links without anchoring

### PR DESCRIPTION
**Pull request by leeoniya**
(cause it's better)

### CODE REVIEW

File: public/app/plugins/panel/xychart/XYChartTooltip.tsx [hunk 1]
```diff
@@ -83,27 +83,25 @@ export const XYChartTooltip = ({ dataIdxs, seriesIdx, data, allSeries, dismiss,
...
+    const v = yField.values[rowIndex];
+    const disp = yField.display ? yField.display(v) : { text: `${v}`, numeric: +v };
+    links = yField.getLinks({ calculatedValue: disp, valueRowIndex: rowIndex }).map((linkModel) => {
+      if (!linkModel.title) {
+        linkModel.title = getTitleFromHref(linkModel.href);
+      }
+
+      return linkModel;
+    });
+  }
```
Comment (leeoniya) [inline public/app/plugins/panel/xychart/XYChartTooltip.tsx:98]: keeping things uniform with other tooltips. helps copy/paste :rofl:

File: packages/grafana-ui/src/components/VizTooltip/VizTooltipFooter.tsx [hunk 2]
```diff
@@ -33,7 +33,7 @@ export const VizTooltipFooter = ({ dataLinks, annotate }: Props) => {
   return (
     <div className={styles.wrapper}>
       {dataLinks.length > 0 && <div className={styles.dataLinks}>{renderDataLinks()}</div>}
-      {annotate && (
+      {annotate != null && (
```
Comment (leeoniya) [inline packages/grafana-ui/src/components/VizTooltip/VizTooltipFooter.tsx:36]: this is a it faster than coercion from function to boolean. tooltips re-renders a lot, so worth it here.

**Review by baldm0mma — APPROVED** [review]
Spun up locally and works as expected.

File: public/app/plugins/panel/xychart/XYChartTooltip.tsx [hunk 3]
```diff
@@ -83,27 +83,13 @@ export const XYChartTooltip = ({ dataIdxs, seriesIdx, data, allSeries, dismiss,
...
-        if (!linkModel.title) {
-          linkModel.title = getTitleFromHref(linkModel.href);
-        }
-
-        return linkModel;
-      });
-    }
-    return links;
-  };
+  const links = getDataLinks(yField, rowIndex);
```
Comment (baldm0mma) [inline public/app/plugins/panel/xychart/XYChartTooltip.tsx:86]: Nice!

**Pull request merged**